server/repository/src/db_diesel/temperature_chart/README.md

```diff
@@ -0,0 +1,49 @@
+### Aggregated temperature chart
+
+The purpose of this repository is to provide aggregated temperature chart data to front end.
+This work is best outsource to database, to avoid excessive serialisations and use power of DB engine.
+
+## Details
+
+Temperature logs are aggregated across a series of intervals, grouped by sensor id, we want to achieve a query similar to this:
+```sql
+
+SELECT avg(temperature), sensor_id, from_datetime, to_datetime FROM 
```

Comment:
Sorry, I messed up one of the lines in SQL statement:

> not nit picking! struggling to understand. I wondered if it would help to have

> [
>   { interval_1_start_datetime, interval_1_end_datetime },
>   { interval_2_start_datetime, interval_2_end_datetime },
>   { interval_3_start_datetime, interval_3_end_datetime }
> ]

Yeah that's much better, in this commit: a0241453d3b3b16e698c01cfcf8e5082730155da, i did just that, and also mentioned that `interval_2_start_datetime == interval_1_end_datetime` and `interval_3_start_datetime ==interval_2_end_datetime`.

> Also - doesn't the first interval end = the start of the next interval? so it would be simpler to have

I think that's right, but we want to aggregated temperature logs for the interval, so would still need from_datetime and to_datetime to do `JOIN temperature_log ON (datetime >= from_datetime and datetime < to_datetime)` at least in the query, but I agree with your previous comment, that API should only return one datetime, we can map that in service or graphql layer.

Hopefully the updated README looks good now